app/src/main/java/ch/epfl/cs311/wanderwave/model/remote/TrackConnection.kt

```diff
@@ -0,0 +1,43 @@
+package ch.epfl.cs311.wanderwave.model.remote
+
+import ch.epfl.cs311.wanderwave.model.data.Track
+import com.google.firebase.firestore.DocumentSnapshot
+import com.google.firebase.firestore.FirebaseFirestore
+
+class TrackConnection(private val database: FirebaseFirestore? = null) :
+    FirebaseConnection<Track, Track>() {
+
+  // THe goal is to have the Id of the firebase document to match the id of the spotify track
+
+  override val collectionName: String = "tracks"
+
+  override val getItemId = { track: Track -> track.id }
+
+  override val db = database ?: super.db
+
+  // Document to Track
+  override fun documentToItem(document: DocumentSnapshot): Track? {
+    return Track.from(document)
+  }
+
+  override fun itemToMap(track: Track): Map<String, Any> {
+    return track.toMap()
+  }
+
+  fun addList(tracks: List<Track>) {

```

Comment:
Should we even expose a function like that?
I can't think of any reason where this would be preferable to `addItemsIfNotExist()`

Reply:
@KlaKalma

Reply:
Sure

Reply:
Deleted